mcp-sqlServer/src/services/kusto_mcp_tools.py

```diff
@@ -0,0 +1,82 @@
+"""
+KQL Generator MCP tool
+Generates KQL queries based on sample templates and user questions through MCP protocol
+"""
+import os
+from datetime import datetime
+from typing import Any, Dict
+
+class KQLGeneratorMCP:
+    """KQL Generator MCP tool for template-based query generation"""
+    
+    def __init__(self):
+        self.sample_kql_path = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 
+            "reference", "samples", "sample.kql"
+        )
+    
+    async def generate_kql_from_template(self, user_question: str) -> Dict[str, Any]:
+        """
+        Based on sample.kql template and user question, return structured data for AI processing
```

Comment:
Think about when will kusto tool be triggered? We should let agent call kql tool if the data is not found in SQL server